### PR TITLE
Attempt to improve RBC completion checking

### PR DESCRIPTION
**Pull request by Vagabond**

### CODE REVIEW

**Comment by Vagabond** [comment]
I think this is ready for a look @evanmcc @vihu

File: test/hbbft_distributed_SUITE.erl [hunk 1]
```diff
@@ -215,6 +218,207 @@ serialization_test(Config) ->
     [ unlink(W) || {_, {ok, W}} <- Workers ],
     ok.
 
+partition_test(Config) ->
```
Comment (evanmcc) [inline test/hbbft_distributed_SUITE.erl:221]: these two tests are highly duplicative.  the filtering is subtle (it took me a couple of reads to even find it), so maybe call that out more and explain what you're trying to provoke with it.  overall the 'why' of the tests is vague, so maybe a block comment above both explaining the strategy employed and the kinds of issues you're trying to provoke?

File: test/hbbft_relcast_distributed_SUITE.erl [hunk 2]
```diff
@@ -136,3 +136,234 @@ simple_test(Config) ->
 
     [ unlink(W) || {_, {ok, W}} <- Workers ],
     ok.
+
```
Comment (evanmcc) [inline test/hbbft_relcast_distributed_SUITE.erl:139]: are these just dupes of the other stuff but run through the relcast worker?

File: src/hbbft_rbc.erl [hunk 3]
```diff
@@ -209,40 +193,50 @@ check_completion(Data = #rbc_data{n=N, f=F}, H) ->
...
-                                    {Data#rbc_data{state=waiting, msg=Msg}, ok}
-                            end;
-                        false ->
-                            %% send ready(h)
-                            {Data#rbc_data{state=waiting, msg=Msg, ready_sent=true}, {send, [{multicast, {ready, H}}]}}
+    Echoes = maps:get(H, Data#rbc_data.num_echoes, []),
+    Readies = maps:get(H, Data#rbc_data.num_readies, []),
+
+    case {length(Echoes), length(Readies)} of
+        {NumEchoes, NumReadies} when NumEchoes >= N - 2*F andalso NumReadies >= 2*F + 1 ->
```
Comment (vihu) [inline src/hbbft_rbc.erl:200]: I'm wondering if the order of `NumEchoes >= N - 2*F` vs `NumReadies >= 2*F + 1` matters here?

Comment (Vagabond) [reply src/hbbft_rbc.erl:200]: the order should not matter, at least from how I read the protocol description quoted immediately below

File: test/hbbft_relcast_distributed_SUITE.erl (hunk 2 above)
Comment (Vagabond) [reply test/hbbft_relcast_distributed_SUITE.erl:139]: yes

File: src/hbbft_rbc.erl (hunk 3 above)
Comment (Vagabond) [reply src/hbbft_rbc.erl]: Addressed

**Pull request merged**